### PR TITLE
Add waffle switch to disable all legacy extension/theme submissions/updates

### LINKED ISSUE #9147: Disable legacy extension version submissions
On September 6th, we need to stop accepting new legacy extension version submissions. Developers will be notified in advance of this event.

In the upload step of the submission process, if a legacy extension is detected, an error should be shown with the message: "Legacy extensions are no longer supported in Firefox."

This applies to both listed and unlisted versions, whether they are new add-ons or updates. It only applies to extensions, since legacy dictionaries are still accepted. The admin override in the upload step should still work.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #9147

### CODE REVIEW

File: src/olympia/devhub/tasks.py [hunk 1]
```diff
@@ -369,6 +369,15 @@ def annotate_legacy_addon_restrictions(results, is_new_upload):
         insert_validation_message(
             results, message=msg, msg_id='thunderbird_and_seamonkey_migration')
 
+    # Legacy add-ons are no longer supported on AMO (if waffle is enabled).
+    elif (is_extension_or_complete_theme and
+            waffle.switch_is_active('disallow-legacy-submissions')):
+        msg = ugettext(
+            u'Legacy extensions are no longer supported in Firefox.')
```
Comment (EnTeQuAk) [inline src/olympia/devhub/tasks.py]: Should this link to our blog maybe, for extra guidance?

Comment (eviljeff) [reply src/olympia/devhub/tasks.py]: Up to @jvillalobos - the issue didn't specify a link to include.

File: src/olympia/devhub/tasks.py [hunk 2]
```diff
@@ -369,6 +369,15 @@ def annotate_legacy_addon_restrictions(results, is_new_upload):
         insert_validation_message(
             results, message=msg, msg_id='thunderbird_and_seamonkey_migration')
 
+    # Legacy add-ons are no longer supported on AMO (if waffle is enabled).
+    elif (is_extension_or_complete_theme and
```
Comment (diox) [inline src/olympia/devhub/tasks.py]: Actually... I don't think there is any other type we'd want to accept.

Comment (eviljeff) [reply src/olympia/devhub/tasks.py]: search plugins?  (dictionaries and language packs are webext now?)

Comment (diox) [reply src/olympia/devhub/tasks.py]: mm yeah search plugins would still need to be allowed. But legacy language packs or dictionaries shouldn't be accepted anymore with this waffle enabled IMHO.

Comment (eviljeff) [reply src/olympia/devhub/tasks.py]: @jvillalobos the issue only specified extensions and complete themes - are other addon types to be blocked too?

**Comment by eviljeff** [comment]
Actually, as this is written as @jvillalobos write in the issue I'm going to merge.  If he second guesses himself we can have a followup issue 😜

File: src/olympia/devhub/tasks.py (hunk 1 above)
Comment (jvillalobos) [reply src/olympia/devhub/tasks.py]: We emailed the affected developers about it, so I don't think it'll be much use.

File: src/olympia/devhub/tasks.py (hunk 2 above)
Comment (jvillalobos) [reply src/olympia/devhub/tasks.py]: I think language packs is covered by https://github.com/mozilla/addons-linter/issues/1985. Legacy dictionaries are still allowed (didn't make it to ESR 60, per [bug 1410214](https://bugzilla.mozilla.org/show_bug.cgi?id=1410214)). Search plugins are still allowed for now.

**Pull request merged**